File: Week 4/DemoDistributedObject/DemoDistributedObject/ServerObjectManager.cs
using System;
using System.Collections.Generic;

namespace DemoDistributedObject
{
    internal class ServerObjectManager
    {
        private static int NextAvailableHandle=1;
        private static Dictionary<int, SObject> objects 
            = new Dictionary<int, SObject>();

        internal static int RegisterObject(SObject sObject)
        {
            sObject.Handle = NextAvailableHandle++;
            objects.Add(sObject.Handle, sObject);
            return sObject.Handle;
        }

        internal static int CreateObject(string strClassName)
        {
            SObject obj = null;
            switch (strClassName)
            {
                case "SStudent":
                    obj = new SStudent();
                    break;
                case "SCourse":
                    obj = new SCourse();
                    break;
            }
            return obj.Handle;
        }

        private static SObject FindObjectByHandle(int handle)
        {
            if (objects.ContainsKey(handle))
                return objects[handle];
            return null;
        }

        public static bool SetObjectAttribute(int handle, 
            string sAttributeName, 
            string newValue)
        {
            try
            {
                SObject obj = FindObjectByHandle(handle);
                return obj.SetAttribute(sAttributeName, newValue);
            }
            catch(Exception)
            {
                return false;
            }
            return true;
        }


        public static string GetObjectAttribute(int handle,
            string sAttributeName)
        {
            try
            {
                SObject obj = FindObjectByHandle(handle);
                return obj.GetAttribute(sAttributeName);
            }
            catch (Exception)
            {
                return "";
            }
            return "";
        }




    }
}

File: Week 4/Java code/Middleware/src/server/ServerObjectManager.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package server;

import java.util.HashMap;
import java.util.Map;

/**
 *
 * @author chungnt
 */
public class ServerObjectManager {

    private static int NextAvailableHandle = 1;

    private static Map<Integer, SObject> objects = new HashMap<Integer, SObject>();

    public static int registerObject(SObject sObject) {
        sObject.handle = NextAvailableHandle++;
        objects.put(sObject.handle, sObject);
        return sObject.handle;
    }

    public static int createObject(String strClassName) {
        SObject sObject = null;
        switch (strClassName) {
            case "CProduct":
                sObject = new SProduct();
                break;
        }
        if (sObject != null) {
            registerObject(sObject);
            return sObject.handle;
        }
        return -1;
    }

    public static SObject findObjectByHandle(int handle) {
        if (objects.containsKey(handle)) {
            return objects.get(handle);
        }
        return null;
    }

    public static boolean setObjectAttribute(int handle, String sAttributeName, String newValue) {
        try {
            SObject sObject = findObjectByHandle(handle);
            return sObject.setAttribute(sAttributeName, newValue);
        } catch (Exception ex) {
            System.err.println(ex.getMessage());
            return false;
        }
    }
    
    public static String getObjectAttribute(int handle, String sAttribute){
        try{
            SObject sObject = findObjectByHandle(handle);
            return sObject.getAttribute(sAttribute);
        }catch(Exception ex){
            System.err.println(ex.getMessage());
            return "";
        }
    }
}